planner/core/fragment.go

```diff
@@ -312,10 +363,19 @@ func (e *mppTaskGenerator) generateMPPTasksForFragment(f *Fragment) (tasks []*kv
 		for _, r := range f.ExchangeReceivers {
 			childrenTasks = append(childrenTasks, r.Tasks...)
 		}
+		cteProducerTasks := make([]*kv.MPPTask, 0)
+		for _, cteR := range f.CTEReaders {
+			child := cteR.children[0]
+			if _, ok := child.(*PhysicalProjection); ok {
+				child = child.Children()[0]
+			}
+			cteProducerTasks = append(cteProducerTasks, child.(*PhysicalExchangeReceiver).Tasks...)
```

Comment:
CTEReaders(PhysicalCTE ) can only be one of the two? possible to assert failure?
* PhysicalProjection->PhysicalExchangeReceiver
* PhysicalExchangeReceiver

Reply:
oo, PhysicalProjection is made from nullable alignment, understood